### PR TITLE
feat(button-bar): follow accessibility behavior

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
https://jira.refinitiv.com/browse/ELF-1721

https://www.w3.org/TR/wai-aria-practices/examples/toolbar/toolbar.html

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by Sarin-Udompanish — APPROVED** [review]

File: packages/elements/src/button/__test__/button.test.js [hunk 1]
```diff
@@ -242,6 +256,12 @@ describe('button/Button', () => {
         ignoredRules: ['aria-allowed-attr', 'color-contrast']
       });
     });
+
+    it('aria-checked should exist, when role=radio', async () => {
```
[the remark this answers is not in the document]
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/button/__test__/button.test.js]: fixed

File: packages/elements/src/button-bar/index.ts [hunk 2]
```diff
@@ -103,6 +142,131 @@ export class ButtonBar extends BasicElement {
...
+      default:
+        return;
+    }
+
+    event.preventDefault();
+  }
+
+  /**
+   * Navigate to next or previous focusable button
+   * @param direction up/next; down/previous
```
[the remark this answers is not in the document]
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/button-bar/index.ts]: fixed

File: packages/elements/src/button-bar/index.ts [hunk 3]
```diff
@@ -95,6 +111,29 @@ export class ButtonBar extends BasicElement {
   @query('slot:not([name])')
   private defaultSlot!: HTMLSlotElement;
 
+  /**
+   * Mutation Observer used to detect changes in the Light DOM
```
[the remark this answers is not in the document]
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/button-bar/index.ts]: Removed mutation so the line is gone

File: packages/elements/src/button-bar/index.ts [hunk 4]
```diff
@@ -95,6 +111,29 @@ export class ButtonBar extends BasicElement {
...
+  private mutationObserver = new MutationObserver(() => this.manageTabIndex());
+
+  /**
+   * Called when connected to DOM
+   * @returns {void}
+   */
+  public connectedCallback (): void {
+    super.connectedCallback();
+    this.mutationObserver.observe(this, observerOptions);
+    !isIE && this.manageTabIndex(); // In IE the mutation will trigger
```
[the remark this answers is not in the document]
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/button-bar/index.ts]: Mutation Observer is removed so the line is gone.

File: packages/elements/src/button-bar/__demo__/index.html [hunk 5]
```diff
@@ -122,5 +122,20 @@
...
+        <ef-button-bar managed role="radiogroup" aria-label="Text Alignment">
+          <ef-button toggles role="radio" icon="text-left" aria-label="left Alignment"></ef-button>
+          <ef-button toggles role="radio" icon="text-center" aria-label="center Alignment"></ef-button>
+          <ef-button toggles role="radio" icon="text-right" aria-label="right Alignment"></ef-button>
+        </ef-button-bar>
+        <ef-button-bar>
+          <ef-button disabled>Copy</ef-button>
+          <ef-button aria-label="paste" icon="paste" readonly></ef-button>
+        </ef-button-bar>
+      </ef-button-bar>
```
Comment (wsuwt) [inline packages/elements/src/button-bar/__demo__/index.html:138]: Can you also make the all demo examples in `button-bar` accessible ?

File: packages/elements/src/button-bar/index.ts [hunk 6]
```diff
@@ -29,6 +29,11 @@ export class ButtonBar extends BasicElement {
     return VERSION;
   }
 
+  /**
+   * Element's role attribute for accessibility
```
Comment (wsuwt) [inline packages/elements/src/button-bar/index.ts:33]:
```suggestion
   * Element's role attribute for accessibility
   * `role` should be `radiogroup` when it is managed.
```

File: packages/elements/src/button-bar/index.ts [hunk 7]
```diff
@@ -103,6 +107,135 @@ export class ButtonBar extends BasicElement {
...
+        this.onTapHandler(event as unknown as TapEvent);
+        break;
+      case 'Right':
+      case 'ArrowRight':
+        // Prevent calling twice if this component is nested
+        !this.isNested() && this.navigateToSibling('next');
+        break;
+      case 'Down':
+      case 'ArrowDown':
+        // Managed works as role radiogroup so UP/DOWN navigate among radios in the group
```
Comment (wsuwt) [inline packages/elements/src/button-bar/index.ts:137]:
```suggestion
        // Managed works as role radiogroup so `Up` and `Down` key can navigate among radios in the group
```

File: packages/elements/src/button-bar/index.ts [hunk 8]
```diff
@@ -103,6 +107,135 @@ export class ButtonBar extends BasicElement {
...
+    this.rovingTabIndex(nextButton, buttons);
+  }
+
+  /**
+   * Navigate to the first focusable button
+   * @returns {void}
+   */
+  private first (): void {
+    const buttons = this.getFocusableButtons();
+    buttons[0].focus();
```
Comment (wsuwt) [inline packages/elements/src/button-bar/index.ts:187]: What happens when there is no button at all ?

File: packages/elements/src/button-bar/index.ts [hunk 9]
```diff
@@ -103,6 +107,135 @@ export class ButtonBar extends BasicElement {
...
+    this.rovingTabIndex(buttons[0], buttons);
+  }
+
+  /**
+   * Navigate to the last focusable button
+   * @returns {void}
+   */
+  private last (): void {
+    const buttons = this.getFocusableButtons();
+    buttons[buttons.length - 1].focus();
```
Comment (wsuwt) [inline packages/elements/src/button-bar/index.ts:197]: What happens when there is no button at all ?

File: packages/elements/src/button-bar/index.ts [hunk 10]
```diff
@@ -103,6 +107,135 @@ export class ButtonBar extends BasicElement {
...
+    this.rovingTabIndex(buttons[buttons.length - 1], buttons);
+  }
+
+  /**
+   * Sets the tabindex to -1 for all buttons except the currently focused one.
+   * @param target the button to be focused
+   * @param buttons Array of Buttons that contains target
+   * @returns {void}
+   */
+  private rovingTabIndex (target: Button, buttons: Button[]):void {
```
Comment (wsuwt) [inline packages/elements/src/button-bar/index.ts:207]:
```suggestion
  private rovingTabIndex (target: Button, buttons: Button[]): void {
```

File: packages/elements/src/button-bar/index.ts [hunk 11]
```diff
@@ -103,6 +107,135 @@ export class ButtonBar extends BasicElement {
...
+      button.tabIndex = -1;
+    });
+    target.tabIndex = 0;
+  }
+
+  /**
+   * Set tabIndex to all buttons
+   * @returns {void}
+   */
+  private manageTabIndex ():void {
```
Comment (wsuwt) [inline packages/elements/src/button-bar/index.ts:218]:
```suggestion
  private manageTabIndex (): void {
```

**Review by wsuwt — APPROVED** [review]

**Comment by wsuwt** [comment]
When it is ready, please squash and use improve screen reader support as commit message.

**Pull request merged**